File: leetcode/1.py
#class Solution:
#    def twoSum(self, nums: List[int], target: int) -> List[int]:
#        for i, a in enumerate(nums):
#            for j, b in enumerate(nums):
#                if i != j and a + b == target:
#                    return [i, j]
#        
#        return -1

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        for i, a in enumerate(nums):
            match = target - a
            if match in nums:
                j = nums.index(match)
                if i != j:
                    return [i, j]
        return -1

File: leetcode/1.rs
impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        for (i, &a) in nums.iter().enumerate() {
            for (j, &b) in nums.iter().enumerate().skip(i+1) {
                if a + b == target {
                    return vec![i as i32, j as i32];
                }
            }
        }
        vec![]
    }
}
